fetch/mssqlserver/ercoleAgentMsSQLServer-Gather.ps1

```diff
@@ -0,0 +1,273 @@
+                    if ($($dbInfo.state_desc) -eq 'ONLINE'){
+                        $dbQueryList| ForEach-Object{
+                            $dbItemList = $_;
+                            $queryPath = $($dbItemList.queryPath);
+                            $outputType = $($dbItemList.outputType); 
+                            $outputFile = $( $($dbItemList.outputFile) +'_'+ $($dbInfo.name) +'.'+ $outputType );
+                            
+                            $parameters = @{
+                                pdatabase_id =  @{name='pdatabase_id'; value=$($dbInfo.database_id); datatype='Int'; size=$null; precision=$null}
+                            }	
+                            if ($outputAs -eq "file"){
+                                .\ercoleAgentMsSQLServer-Query.ps1 -instance $instance -dbName $($dbInfo.name) -QueryPath $queryPath -OutputFile $outputFile -OutputType $outputType -Parameters $parameters -outputAs $outputAs
+                            }else{
+                                
+                                $execResult = .\ercoleAgentMsSQLServer-Query.ps1 -instance $instance -dbName $($dbInfo.name) -QueryPath $queryPath -OutputFile $outputFile -OutputType $outputType -Parameters $parameters -outputAs object|ConvertFrom-Json
+                                $auxLayout = $($dbItemList.outputFile)
+                                $auxLayout = $auxLayout.Substring($auxLayout.LastIndexOf('\')+1, $auxLayout.Length - $auxLayout.LastIndexOf('\')-1)
+                                $jsonbase = @{layout=$auxLayout; database_name=$($dbInfo.name)}
+                                $jsonbase.Add("data",$execResult)
+                                $resultList.Add($jsonbase) |Out-Null
+                            }
+                     }
+                    }else{
+                        Write-Output $('Alert:' + $($dbInfo.name) +'is '+ $($dbInfo.state_desc)) |Out-File -FilePath $("$outDir"+'gather.log') -Append
+                    }
+                }
+                if ($outputAs -eq "object"){
+                    #$jsonbase = @{layout=;data=$outputList}
+                    #$jsonbase| ConvertTo-Json -Depth 10
+                    $resultList| ConvertTo-Json -Depth 10
+                }
+            }
+        }
+    }
+}
+
+
+ function installedPatches(){
+    #Installed Patches
+    $jsonbase = @{layout='patches'}
+    $execResult = Get-ChildItem -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall | Get-ItemProperty | Sort-Object -Property DisplayName `
+        | Select-Object -Property DisplayName, DisplayVersion, InstallDate `
+        | Where-Object {($_.DisplayName -like "Hotfix*SQL*") -or ($_.DisplayName -like "Service Pack*SQL*")} 
+
+    $jsonbase.Add("data",$execResult)                            
+    if ($outputAs -eq "file"){
+        $jsonbase | ConvertTo-Json -Depth 10|Out-File -FilePath $("$outDir"+'patches.json')
```

Comment:
It's necessary to write files? Can you instead save the stdout in a variable or write to a temporary file?